src/generic/RationalFunctionField.jl

```diff
@@ -475,17 +475,17 @@ end
 ###############################################################################
 
 function zero!(c::RationalFunctionFieldElem)
-   c.d = zero!(data(c))
+   zero!(data(c))
    return c
 end
 
 function mul!(c::RationalFunctionFieldElem{T, U}, a::RationalFunctionFieldElem{T, U}, b::RationalFunctionFieldElem{T, U}) where {T <: FieldElement, U <: Union{PolyRingElem, MPolyRingElem}}
-   c.d = mul!(data(c), data(a), data(b))
+   mul!(data(c), data(a), data(b))
```

Comment:
1. Will this work if we make the type of `.d` immutable?
2. Do we need to rewrite these `!` functions to `mul!(c, a, b) = RationalFunctionField(parent(c), mul!(data(c), data(a), data(b))` to not be dependable on whether the fields themselves are mutable?

Reply:
Your suggestion to use `mul!(c, a, b) = RationalFunctionField(parent(c), mul!(data(c), data(a), data(b))` is much better, and basically at least as efficient because `RationalFunctionFieldElem` is now immutable and so can be e.g. stack allocated.

I'll update the PR